x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/ProjectOperator.java

```diff
@@ -67,8 +67,10 @@ protected Page process(Page page) {
             }
             var block = page.getBlock(source);
             blocks[b++] = block;
+            block.incRef();
         }
-        return page.newPageAndRelease(blocks);
+        page.releaseBlocks();
+        return new Page(blocks);
```

Comment:
Another simplification: just close all the blocks and all is good.

Reply:
Maybe this can be generalized by putting this into the driver, rather than having the operator care about releasing blocks.